controllers/signup.js

```diff
@@ -0,0 +1,22 @@
+const signUp = require("../public/javascripts/data/signup");
+
+
+// GET request
+exports.signUp = function(req, res, next) {
+  res.render('signup');
+}
+
+// POST request
+exports.signUpUser = function(req, res, next) {
+  let username = req.body.username;
+  let password = req.body.password;
+  if(signUp.signUp(username)) {
```

Comment:
I'd negate the if check here too. Check the failure condition and exit early inside the if block, let the success condition continue outside of the if conditional